src/phase0/state_transition/block_processing.rs

```diff
@@ -160,16 +163,38 @@ fn process_block_header<
         MAX_VALIDATORS_PER_COMMITTEE,
         PENDING_ATTESTATIONS_BOUND,
     >,
-    _block: &mut BeaconBlock<
+    block: &mut BeaconBlock<
```

Comment:
this `&mut` was an accident in https://github.com/ralexstokes/ethereum_consensus/commit/68a652bbfc269ff142c7b36e97c78b6d7ae68eaa and should just be `&`. 

i just fixed it on `main` via #35 which you can pull into this branch if you rebase

Reply:
Done. 
But now there is an issue with `block.body.hash_tree_root()` since `hash_tree_root()` takes `self` as `&mut`. I need clues on how can I work around this issue.
I don't understand, why `hash_tree_root()` wants mutable reference?

Reply:
ah! sorry i mislead you here....

i did think the mutability on `block` was an accident (bc i had just blindly copied it the first time) but we do actually need it for the reason you point you -- taking the `hash_tree_root`.

and the reason why we have `hash_tree_root(&mut self)` is that SSZ types have internal caching that may need to update internal state when getting the hash tree root (edits to some SSZ type do not immediately update the backing Merkle tree to avoid paying the hashing cost if you don't need it)

i have a WIP PR of caching for `Vector`s here if you want to see more: https://github.com/ralexstokes/ssz_rs/pull/6

Reply:
Got it, thank you! Used `&mut`: b7cd5e2